test/states/test.Midroll.js

```diff
@@ -0,0 +1,50 @@
+import QUnit from 'qunit';
+import {Midroll} from '../../src/states.js';
+import adBreak from '../../src/adBreak.js';
+
+/*
+ * These tests are intended to be isolated unit tests for one state with all
+ * other modules mocked.
+ */
+QUnit.module('Midroll', {
+  beforeEach: function() {
+    this.player = {
+      ads: {
+        _inLinearAdMode: false,
+        endLinearAdMode: () => {
+          this.calledEndLinearAdMode = true;
+        }
+      }
+    };
+
+    this.midroll = new Midroll(this.player);
+
+    this.adBreakStartStub = sinon.stub(adBreak, 'start');
+    this.adBreakEndStub = sinon.stub(adBreak, 'end');
+  },
+
+  afterEach() {
+    this.adBreakStartStub.restore();
+    this.adBreakEndStub.restore();
+  }
+});
+
+QUnit.test('starts an ad break on init', function(assert) {
+  this.midroll.init(this.player);
+  assert.equal(this.player.ads.adType, 'midroll', 'ad type is midroll');
+  assert.equal(this.adBreakStartStub.callCount, 1, 'ad break started');
+});
+
+QUnit.test('ends an ad break on endLinearAdMode', function(assert) {
+  this.midroll.init(this.player);
+  this.player.ads._inLinearAdMode = true;
```

Comment:
do you have to hardcode this? could we do this as part of the stub?

Reply:
I wanted the stub to reflect an "initial" state and have each test advance it forward to the necessary state, rather than having the stub be kind of a random assortment of properties.

Reply:
I see. Would calling `startLinearAdMode` be equivalent then?

Reply:
Yeah, originally I wanted to limit the test to only calling one function, but looking back on it now, using startLinearAdMode is a more solid test.

Reply:
My mistake. startLinearAdMode is not defined in Midroll.js, so that's not an option.

Reply:
Ah, this.player.ads._inLinearAdMode is always mocked in this test file. I suppose that is fine then

Reply:
Ah! Just realized that when you enter the Midroll state, _inLinearAdMode would always be true, so it make sense for this to change in the mock. We've gone full circle on this one.

Reply:
Updated